.semgrep/adapter/bid-type-if-check.yml

```diff
@@ -0,0 +1,21 @@
+rules:
+  - id: bid-type-if-check
+    message: Prebid server expects media type set in adapter response. we recommend implementing a pattern where adapter server sets the [MType](https://github.com/prebid/openrtb/blob/53ac1b9ca7ddf860988ff69bf5018b32ced1fbc5/openrtb2/bid.go#L334) field in the response to accurately determine the media type for the impression.
+    languages:
+      - go
+    severity: WARNING
+    patterns:
+      - pattern-inside: |
+          if $CONDITION != nil {
+            return $ORTBTYPE, nil
+          }
+      - focus-metavariable: $CONDITION
+      - metavariable-pattern:
+          metavariable: $ORTBTYPE
+          patterns:
+            - pattern-either:
+                - pattern: openrtb_ext.$W
+                - pattern: imp.$W
```

Comment:
may be we can remove `pattern: imp.$W`

Reply:
Updated in [8c302f5](https://github.com/prebid/prebid-server/pull/2965/commits/8c302f5e7df5ce20b00e213225a45f89bce549f6)